implementations/rust/transport/src/tcp.rs

```diff
@@ -151,11 +150,25 @@ impl TcpManager {
 
             // check for receives
             for a in &self.addresses {
+                let mut l = 0;
                 match self.connections.get_mut(&a.to_string()) {
                     Some(t) => {
-                        if let Err(s) = t.receive_message() {
-                            println!("tcp receive failed {}", s);
-                            return false;
+                        loop {
+                            l += 1;
+                            match t.try_receive() {
```

Comment:
should we use a timeout rather than returning immediately?

Reply:
We could, but I don't like blocking for any time in the case where we're running on a single-threaded system. We'll try again at the next polling interval.